api/v1beta1/config_types.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cfg "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+)
+
+type Feature struct {
+	Enabled bool `json:"enabled"`
+}
+
+type ModuleReconcilerFeature struct {
+	Feature `json:",inline"`
+
+	// If true, allows KMM to build images.
+	AllowBuilds bool `json:"allowBuilds"`
+
+	// If true, allows KMM to start kmod signing jobs.
+	AllowSigning bool `json:"allowSigning"`
+}
+
+type Reconcilers struct {
+	// ManagedClusterModule contains settings for the ManagedClusterModule reconciler.
```

Comment:
We have essentially 3 modes:
1) Regular Module
2) Hub
3) Spoke
Why not just define those 3 flags, and then initialize the reconcilers in the main according to those flags? Having flag for each reconciler is a little bit redundant and confusing: we cannot have ManagedClusterModule false and NodeKernelClusterClaim true and visa-versa, no? WDYT?

Reply:
That was the spirit of #203. But instead of defining those modes in the code, we will create 3 bundles which will enable or disable the appropriate reconcilers for each use case. I think that's more flexible and future-proof.

Reply:
i agree this is more flexible, but the code becomes much less readable: newcomers, or somebody not familiar with the one of the modes won't be able to understand from the code which controllers are needed in which mode.